dash/development/_r_components_generation.py

```diff
@@ -159,35 +166,25 @@ def generate_class_string(name, props, project_shortname, prefix):
     wildcards = ""
     wildcard_declaration = ""
     wildcard_names = ""
+    default_paramtext = ""
+    default_argtext = ""
+    accepted_wildcards = ""
 
     if any("-*" in key for key in prop_keys):
+        accepted_wildcards = get_wildcards_r(prop_keys)
         wildcards = ", ..."
-        wildcard_declaration = (
-            "\n    wildcard_names = names(assert_valid_wildcards(...))\n"
-        )
+        wildcard_declaration = """
+    wildcard_names = names(dash_assert_valid_wildcards(attrib = list({}), ...))
+""".format(accepted_wildcards.replace("-*", ""))       # noqa:E501
         wildcard_names = ", wildcard_names"
 
-    default_paramtext = ""
-    default_argtext = ""
-    default_wildcards = ""
-
     # Produce a string with all property names other than WCs
     prop_names = ", ".join(
         "'{}'".format(p)
         for p in prop_keys
         if "*" not in p and p not in ["setProps"]
     )
 
-    # in R, we set parameters with no defaults to NULL
-    # Here we'll do that if no default value exists
-    default_wildcards += ", ".join("'{}'".format(p)
-                                   for p in prop_keys if "*" in p)
-
-    if default_wildcards == "":
-        default_wildcards = "NULL"
-    else:
-        default_wildcards = "c({})".format(default_wildcards)
-
     # Filter props to remove those we don't want to expose
     for item in prop_keys[:]:
         if item.endswith("-*") or item in r_keywords or item == "setProps":
```

Comment:
also not new in this PR, but `item in r_keywords` worries me to just be silently dropped - if nothing else it seems like this should log a warning. Do any of our own components use R keywords?

Reply:
Added a warning if we drop something from `r_keywords`, in https://github.com/plotly/dash/pull/783/commits/e35f040c112dd36642b9d1b235781a8d3451e130

We should keep an eye out for this when rebuilding all the components - if we never see this warning, perhaps we can lock the R keywords out by turning the warning into an error.